src/Tribe/Repositories/Venue.php

```diff
@@ -0,0 +1,181 @@
+<?php
+/**
+ * The main ORM/Repository class for venues.
+ *
+ * @since TBD
+ */
+
+/**
+ * Class Tribe__Events__Repositories__Venue
+ *
+ *
+ * @since TBD
+ */
+class Tribe__Events__Repositories__Venue extends Tribe__Repository {
+
+	/**
+	 * The unique fragment that will be used to identify this repository filters.
+	 *
+	 * @var string
+	 */
+	protected $filter_name = 'venues';
+
+	/**
+	 * Tribe__Events__Repositories__Venue constructor.
+	 *
+	 * Sets up the repository default parameters and schema.
+	 *
+	 * @since TBD
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		$this->default_args = array(
+			'post_type'                    => Tribe__Events__Venue::POSTTYPE,
+			// We'll be handling the dates, let's mark the query as a non-filtered one.
+			'tribe_suppress_query_filters' => true,
+		);
+
+		$this->schema = array_merge( $this->schema, array(
+			'address'        => array( $this, 'filter_by_address' ),
+			'city'           => array( $this, 'filter_by_city' ),
+			'country'        => array( $this, 'filter_by_country' ),
+			'phone'          => array( $this, 'filter_by_phone' ),
+			'postal_code'    => array( $this, 'filter_by_postal_code' ),
+			'province'       => array( $this, 'filter_by_province' ),
+			'state'          => array( $this, 'filter_by_state' ),
+			'state_province' => array( $this, 'filter_by_state_province' ),
+			'website'        => array( $this, 'filter_by_website' ),
+		) );
+	}
+
+	/**
+	 * Filter by LIKE or REGEX string.
+	 *
+	 * @param string $meta_key Meta key to filter by.
+	 * @param string $value    MySQL compatible LIKE or REGEX string.
+	 */
+	public function filter_by_like_regex( $meta_key, $value ) {
+		if ( tribe_is_regex( $value ) ) {
+			$this->by( 'meta_regexp', $meta_key, tribe_unfenced_regex( $value ) );
+
+			return;
+		}
+
+		$this->by( 'meta_like', $meta_key, $value );
+	}
+
+	/**
+	 * Filters venues by a specific address line.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $value MySQL compatible LIKE or REGEX string.
```

Comment:
Here and in any other place where it appears: MySQL uses `REGEXP` with a final `P`. 
If we note it to refer to SQL use `REGEXP`, not `REGEX`.

Reply:
Copy/pasted from the spec doc, will update